core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -245,6 +248,13 @@ private void tryStart(Profiler profiler) {
             applyConfiguration(createCommand);
 
             containerId = createCommand.exec().getId();
+
+            PortForwardingContainer.INSTANCE.getNetwork().map(ContainerNetwork::getNetworkID).ifPresent(networkId -> {
+                if (!Arrays.asList(networkId, "none", "host").contains(createCommand.getNetworkMode())) {
```

Comment:
`host` & `none` networks are not connectable, this is why I added it :)

We could also use a random password to improve the security

Reply:
If adding a random password isn't a massive change, let's do it. I can't think of a practical attack via this container, but we may as well just in case we've missed something.

Reply:
changed to generate a random password